### ISSUE TITLE
[Q] Report Learning Rate is always same with HuggingFace Trainer?

### ISSUE THREAD

**Issue by Leo-T-Zang** [wandb/wandb#5941]
why learning rate report ot wandb is always 0.0005 even I change lr

```
lr = 1e-4
wandb.init( config = {"learning_rate": lr})
training_args = TrainingArguments(
    output_dir='./output/',
    num_train_epochs = 5,
    per_device_train_batch_size = 8,
    per_device_eval_batch_size = 8,
    warmup_steps = 300,
    logging_dir='./logs',
    logging_steps=10,
    evaluation_strategy="epoch",
    load_best_model_at_end=True,
    save_strategy='epoch',
    metric_for_best_model='eval_loss',
    save_total_limit = 5,
    report_to='wandb',
)
train_dataset = Dataset("train.csv", tokenizer)
test_dataset = Dataset("test.csv", tokenizer)

trainer = Trainer(
    model=model,
    args=training_args,
    train_dataset=train_dataset,
    eval_dataset=test_dataset,
    optimizers=(AdamW(model.parameters(), lr=lr), None),
)
trainer.train()
```

**Comment by ArtsiomWB** [comment]
Hi @Leo-T-Zang! Thank you for writing in. When you go into your wandb run config in wandb UI, what learning rate do you see in wandb?

**Comment by Leo-T-Zang** [comment]
Hi @ArtsiomWB, 
What I have seen is always 0.00005 for learning rate.

**Comment by ArtsiomWB** [comment]
Thank you for confirming! Could you send me a link to your workspace?

**Comment by ArtsiomWB** [comment]
Hey Leo! Wanted to follow up on your request.

**Comment by sydholl** [comment]
Artsiom Skarakhod commented: 
Hi @Leo-T-Zang! Thank you for writing in. When you go into your wandb run config in wandb UI, what learning rate do you see in wandb?

**Comment by sydholl** [comment]
Artsiom Skarakhod commented: 
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by Leo-T-Zang** [comment]
Sorry for the late reply. @sydholl  @ArtsiomWB 
Here is the link of my workspace:
https://wandb.ai/tian-lai-zang/PepMLM?workspace=user-tian-lai-zang

Thank you so much! Please let me know if anything else is needed.

**Comment by ArtsiomWB** [comment]
Thank you for sending them over @Leo-T-Zang! 
I do see that for some of the runs, learning rate differs, were those runs just not affected by this concern, and ran as all of the other ones, or dod you use a different config for them?
<img width="99" alt="image" src="https://github.com/wandb/wandb/assets/115654725/dbbe9538-3fee-4ec2-9184-abeb28143b3e">

**Comment by Leo-T-Zang** [comment]
So those have different learning rate is coming from sweep runs.

Here is the sweep code for your reference:
```
def train():
    model = "X"
    tokenizer = XXX
    
    # Initialize the W&B run
    run = wandb.init(project="X")
    config = run.config

    training_args = TrainingArguments(
        output_dir='./output/',
        num_train_epochs = 5,
        per_device_train_batch_size = config.batch_size,
        per_device_eval_batch_size = 8,
        warmup_steps = config.warmup_steps,
        logging_dir='./logs',
        logging_steps=10,
        evaluation_strategy="epoch",
        load_best_model_at_end=True,
        save_strategy='epoch',
        metric_for_best_model='eval_loss',
        save_total_limit = 1,
        report_to='wandb',
        gradient_accumulation_steps=config.gradient_accumulation_steps,
    )

    train_dataset = ProteinDataset("train.csv", tokenizer)
    test_dataset = ProteinDataset("test.csv", tokenizer)

    trainer = Trainer(
        model=model,
        args=training_args,
        train_dataset=train_dataset,
        eval_dataset=test_dataset,
        optimizers=(AdamW(model.parameters(), lr = config.learning_rate), None),
    )

    trainer.train()
    run.finish()

sweep_config = {
    'method': 'bayes',  
    'metric': {
      'name': 'eval_loss',
      'goal': 'minimize'   
    },
    'parameters': {
        'learning_rate': {
            'min': 1e-6,
            'max': 1e-3
        },
        'warmup_steps': {
            'min': 0,
            'max': 1000
        },
        'batch_size': {
            'values': [1, 2, 4, 6, 8]
        },
        'gradient_accumulation_steps': {
            'values': [1, 2, 4, 8, 16]
        }
    }
}

sweep_id = wandb.sweep(sweep_config, project="X")
wandb.agent(sweep_id, function=train)
```

**Comment by ArtsiomWB** [comment]
Perfect! Thank you for sending it over, we are looking into it! I think it might potentially have something to do with hugging face and if they edit their variables (like normalize, or round)

**Comment by ArtsiomWB** [comment]
Hi @Leo-T-Zang! Still no luck reproducing this on my side. Still looking into it!

**Comment by ArtsiomWB** [comment]
Could you try running this training one more time but with wandb disabled, and see if the issue persists?

**Comment by Leo-T-Zang** [comment]
Hi @ArtsiomWB,

Thank you for looking into this problem for such a long time.

I think learning rate is correct for training that I can change any learning rate I want. The problem is when report to Wandb, it just all shows same number. Actually, it first shows up the learning rate I set and after a while, it just changes to 0.0005

**Comment by ArtsiomWB** [comment]
Always happy to help! Could you try setting a different learning rate min and max and see if it still gets stuck for you on the same value over time?

**Comment by ArtsiomWB** [comment]
Hi Leo! Wanted to follow up with you regarding this ticket. Have you had a chance to change the range of of learning rate?

**Comment by Leo-T-Zang** [comment]
Hi @ArtsiomWB, if I use min and max for optimization, the learning rates reported are correct.

**Comment by ArtsiomWB** [comment]
Hi @Leo-T-Zang, that's a start. Would this unblock you for now, or this workflow does not work for your project?

**Comment by ArtsiomWB** [comment]
We wanted to follow up with you regarding your support request as we have not heard back from you. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
Reported by user: Leo-T-Zang
 GitHub URL: https://github.com/wandb/wandb/issues/5941

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
Auto-assigned

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
This ticket has gone stale, please follow up or close.
